File: dotnet/src/Model/Map/Map.cs
using System;
using System.Collections.Generic;

namespace PlanetCS1 {
    class Map {
        List<List<ITile>> tiles {get;set;}
        public Map(int size) {
            this.tiles = new List<List<ITile>>();

            for (int i = 0; i < size; ++i) {
                List<ITile> row = new List<ITile>();
                for (int j = 0; j < size; ++j) {
                    Location location = new Location(i, j);
                    BlankTile blank = new BlankTile(location);
                    row.Add(blank);
                }
                this.tiles.Add(row);
            }
        }
    }
}

File: kotlinjs/old/Map.kt



//  because Kotlin is...silly,
//  we need to use mutable lists to get dynamic-sized maps
class Map(val size: Int) {
    val tiles = mutableListOf<MutableList<ITile>>()

    init {
        clearMap()
    }

    fun clearMap() {
        for (i in 0 until size) {
            val row = mutableListOf<ITile>()
            for (j in 0 until size) {
                val location = Location(i, j)
                val blank = BlankTile(location)
                row.add(blank)
            }
            tiles.add(row)
        }
    }
}